experimental/tools/unit-generators/vhdl/generators/handshake/speculation/speculator.py

```diff
@@ -570,21 +579,16 @@ def _generate_predictor(name, bitwidth):
   begin
     if (rst = '1') then
       data_reg <= zeros & '1';
-      data_out <= zeros & '1';
     elsif (rising_edge(clk)) then
       if (data_in_valid = '1') then
         data_reg <= data_in;
       end if;
-      if (data_out_ready = '1' and trigger_valid = '1') then
-        -- After handshaking, data_out updates and holds its value until the
-        -- next handshaking, ensuring stability while valid is high.
-        data_out <= data_reg;
-      end if;
     end if;
   end process;
 
   data_in_ready <= '1';
 
+  data_out <= data_reg;
```

Comment:
This I reverted the predictor to the original one from Haoran's work.
I was afraid of the predictor not holding the value, but this is now handled together with other specCore's outputs, by the internal TEHB.
The previous implementation introduced delay to the update of predictor's value.